kebechet_administrator.py

```diff
@@ -0,0 +1,149 @@
+    )  # We query without the package_version.
+    for key in repositories.keys():
+        repo_info = repositories[key]
+        # Construct the message input
+        if repo_info.get("private"):
+            continue  # We skip for private repo's.
+        if semver.compare(repo_info.get("package_version"), Configuration.PACKAGE_VERSION):
+            continue  # We dont schedule, if the package version > version of the solved package version.
+        message_input = {
+            "component_name": {"type": "str", "value": __COMPONENT_NAME__},
+            "service_version": {"type": "str", "value": __service_version__},
+            "url": {"type": "str", "value": _URL_PREFIX + key},
+            "service_name": {"type": "str", "value": "github"},
+            "installation_id": {"type": "str", "value": repo_info.get("installation_id")},
+        }
+
+        # We store the message to put in the output file here.
+        output_messages.append({"topic_name": "thoth.kebechet-run-url-trigger", "message_contents": message_input})
+
+
+def _handle_package_issue(Configuration):  # noqa: N803
+    """Handle all the messages for which Kebechet needs to run on all repos associated."""
+    # We getch all the Kebechet repos using a non optimal package(missing package or CVE or missing version)
+    repositories: Dict[str, Dict] = GRAPH.get_kebechet_github_installations_info_for_python_package_version(
+        package_name=Configuration.PACKAGE_NAME,
+        version=Configuration.PACKAGE_VERSION,
+        index_url=Configuration.PACKAGE_INDEX,
+    )
+    # The keys represent the repo names and the value dictionary is the details for the repo.
+    for key in repositories.keys():
+        repo_info = repositories[key]
+        # Construct the message input
+        if repo_info.get("private"):
+            continue  # We skip for private repo's.
+        message_input = {
+            "component_name": {"type": "str", "value": __COMPONENT_NAME__},
+            "service_version": {"type": "str", "value": __service_version__},
+            "url": {"type": "str", "value": _URL_PREFIX + key},
+            "service_name": {"type": "str", "value": "github"},
+            "installation_id": {"type": "str", "value": repo_info.get("installation_id")},
+        }
+
+        # We store the message to put in the output file here.
+        output_messages.append({"topic_name": "thoth.kebechet-run-url-trigger", "message_contents": message_input})
+
+
+# This handler dispatches the specific method based on a paticular message.
+_message_handler = {
+    "SolvedPackageMessage": _handle_solved_message,
+    "HashMismatchMessage": _handle_package_issue,
+    "MissingPackageMessage": _handle_package_issue,
+    "MissingVersionMessage": _handle_package_issue,
+    "CVEProvidedMessage": _handle_package_issue,
+}
+
+
+def _input_validation():
+    required_inputs = frozenset({"PACKAGE_NAME", "PACKAGE_VERSION", "PACKAGE_INDEX", "MESSAGE_TYPE"})
```

Comment:
no check on SOLVER_NAME required for SolvedPackageMessage? otherwise `_handle_solved_message` would raise error when trying to parse SOLVER_NAME

Reply:
Actually MissingPackageMessage does not has package version: https://github.com/thoth-station/messaging/blob/master/thoth/messaging/missing_package.py in this case the entire package disappeared with all releases from the index?, so here we should not re run kebechet with adviser for example, we have to open an issue directly to the repo that has that package in the software stack? WDYT @fridex @saisankargochhayat @KPostOffice

Reply:
> no check on SOLVER_NAME required for SolvedPackageMessage? otherwise _handle_solved_message would raise error when trying to parse SOLVER_NAME

Added the check to sovler handler. Common checks being in input handler

Reply:
For `MissingPackageMessage` that seems like the right behaviour, maybe we want to implement in a different issue and discuss implementation details?

Reply:
@KPostOffice commented today in scrum. WDYT?

Reply:
> Actually MissingPackageMessage does not has package version: https://github.com/thoth-station/messaging/blob/master/thoth/messaging/missing_package.py in this case the entire package disappeared with all releases from the index?, so here we should not re run kebechet with adviser for example, we have to open an issue directly to the repo that has that package in the software stack? WDYT @fridex @saisankargochhayat @KPostOffice

Do we want to add version into the message? What is the semantics behind it? Do we receive multiple messages with all the versions when the whole package disappears?

Reply:
I think one message with just the package name should be fine right? We could act on that message, creating an issue in the user's repo saying that the package has gone missing, we wouldn't need the version, if the whole package has gone missing.

Reply:
Well, there are three issues specifically:

1. removed the whole package
2. removed a specific version of a package
3. removed artifact - e.g. build for py38 but sdist or other wheels are still present

All the three above can cause issues. So we should be able to detect removals of any of these. Any of these might cause the application would not assemble.

Reply:
Thanks @fridex!

> Well, there are three issues specifically:
> 
> 1. removed the whole package

This case can be covered by this case I suppose.

> 2. removed a specific version of a package

We ave `MissingVersionMessage` already and Kebechet admin can take care of that.

> 3. removed artifact - e.g. build for py38 but sdist or other wheels are still present
How do we check this? I guess it should be a new message sent out by package update and investigator and kebechet admin can deal with that. WDYT?

There is also the case of changing hash and that is covered by `HashMismatchMessage` and Kebechet admin can already deal with that with this new logic.

> 
> All the three above can cause issues. So we should be able to detect removals of any of these. Any of these might cause the application would not assemble.